### PR TITLE
Route finding

### PR DESCRIPTION
**Pull request by hc09141**
Adds checks for correct priority type (air, sea and land) matching correct priority to finding best route

### CODE REVIEW

File: app/models/company_route.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@ class CompanyRoute < ActiveRecord::Base
   validates :company, :origin, :destination, :weight_cost, :volume_cost, :max_weight,
             :max_volume, :frequency, :duration, presence: true
   validates :priority_type, presence: true, inclusion: {in: Constants::TYPE}
+  validates :frequency, presence: true
+  validates :duration, presence: true
```
Comment (divyap1) [inline app/models/company_route.rb]: These are in the list above so they can be removed

Comment (hc09141) [inline app/models/company_route.rb]: Must have missed these in the merge conflict, have removed now

File: app/models/create_mail.rb [hunk 2]
```diff
@@ -70,4 +70,20 @@ def generate_best_route(final_edge)
...
+  def calculate_expenditure
+    expenditure = Event.last.expenditure
+    company_routes.each do |route|
+      expenditure -= (route.weight_cost * weight + route.volume_cost * volume)
+    end
+    expenditure.save!
+  end
+
+  def calculate_revenue
+    revenue = Event.last.revenue
```
Comment (Recursively) [inline app/models/create_mail.rb]: You could use tap and inject here to make it `rails way` but I don't think it's overly necessary.

Here is a one liner version:

```ruby
Event.last.revenue.tap { |r| company_routes.inject(Event.last.revenue) { |rev, route| rev + (route.kps_route.weight_price * weight + route.kps_route.volume_cost * volume) }.save!
```

But don't do that.

File: app/controllers/create_mail_controller.rb [hunk 3]
```diff
@@ -15,6 +15,8 @@ def create
     @create_mail_event = CreateMail.new(mail_params)
     @create_mail_event.find_best_route
     if @create_mail_event.company_routes.present?
+      @create_mail_event.calculate_expenditure
```
Comment (Recursively) [inline app/controllers/create_mail_controller.rb]: minor thing, but if a method changes the state of a model, it should use have a `!` on the method name. It's just a rails convention, but again not necessary.

ie:
```ruby
def calculate_expenditure!

@create_mail_event.calculate_expenditure!
```

**Pull request merged**